src/test/java/WordleTest.java

```diff
@@ -0,0 +1,34 @@
+import org.junit.jupiter.api.Test;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class WordleTest
+{
+    @Test
+    public void getDefinition() throws IOException
+    {
+        //given
+        WordleDictionary wordle = new WordleDictionary();
+        //when
+        String word = wordle.getDefinition("BO");
```

Comment:
Write another test but this time call getDefinition() twice. See what happens.

Reply:
it doesn't work, it gives me na out of bounds error

Reply:
Now why does it do that?

Reply:
probably because it only allows me to read through the dictionary once and every new iteration needs to be a new call to the method.

you're gonna tell me to make getDefinition so that it only has to run through the dictionary one time

Reply:
Yes.

Reply:
can you give me some hint of how I would go about this?

Reply:
Use the constructor.